<a href="https://colab.research.google.com/github/ploybongkot/Multivariate/blob/main/python101.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

ตัวแปร variable




หลักการตั้งชื่อตัวแปรเบื้องต้น

1.ตั้งให้สื่อ ว่าตัวแปรนี้คืออะไร เวลาย้อนมาดูจะได้ดูง่ายๆ

2.ภาษาอังกฤษ เพราะภาษาไทยคอมอ่านไม่รู้เรื่อง

3.ใช้ตัวเลขได้แต่ห้ามขึ้นต้นด้วยตัวเลข เช่น ตัวแปรชื่อ a1 a2 a3 ทำได้ แต่ว่าตัวแปรชื่อ 1a ทำไม่ได้

4.ห้ามเว้นวรรค เพราะเว้นวรรคเป็นคำใหม่

5.ตัวเล็กกับตัวใหญ่ไม่เหมือนกัน

มี output 2 mode คือ ออกหน้าจอ ดับ เข้าไปก็บในตัวแปร

In [32]:
3.14159  # hashtag หรือ sharp # คือ comment >>> อันนี้คือ output ที่ออกหน้าจอ

3.14159

In [33]:
Pi = 3.14159  # อันนี้คือ output ที่เข้าไปอยู่ในตัวแปร

In [34]:
Pi

3.14159

In [35]:
a = 1234567
print(a)

1234567


ข้อดีของ code คือ มันสามารถ Enter ลงมาได้ ทำให้เราสามารถพิมพ์คำสั่งยาวๆหลายๆบรรทัดได้

# ชนิดของตัวแปร 3 ชนิด

### จำนวนเต็ม ( integer,int)

In [36]:
aaa = 1092
print(aaa)

1092


### จำนวนจริง (float)

In [37]:
bbb = 11.
print(bbb)

11.0


### ตัวอักษร-ข้อความ (character และ text,string) คือ ค่าหลายๆค่าที่เอาเรียงต่อกัน



In [38]:
ccc = '123456'  # เราใช้ single quote ' หรือ double qoute " ล้อมรอบตัวหนังสือเพื่อระบุว่าเป็น charecter-string ไม่ว่จะเป็น ' หรือ " ใช้ได้หมดเลย แต่ว่าเปิดด้วยตัวไหนก็ต้องปิดด้วยตัวนั้น 
print(ccc)

123456


ถึงเป็นตัวเลขแต่ถ้ามี qoute ล้อมรอบแปลว่าเป็นตัวอักษร หรือ string ไม่ได้เป็นตัวเลข เอาไปใช้บวกลบคูณหารไม่ได้ เหตุผลที่ต้องมี quote ล้อมรอมเพราะว่า ตัวแปรมันก็เป็นตัวอักษรเหมือนกันมันจะได้รู้ว่าอันนี้คือตัวแปล อันนี้คือข้อความไม่ใช่ตัวแปร

In [39]:
aaa + bbb เอา float มาบวกกับ int = float

SyntaxError: ignored

In [40]:
bbb + ccc อันนี้มันจะไม่สามารถบวกได้เพราะว่ามันเป็นตัวแปลคนละชนิดกัน จริงๆตัวแปรคนละชนิดกันก็บวกกันได้

SyntaxError: ignored

ใน python เวลามันทำอะไรไม่ได้มันก็จะ Error เกิดขึ้น จะสนใจแค่บรรทัดสุดท้าย มันจะบอกว่าทำไมถึง Error ถ้าเราอ่านไม่ออกก็สามารถคัดลอกเอาไปหาใน Google ได้ ใน Google จะมีคนอธิบาย Error ที่เกี่ยวกับ Python ไว้หมด

In [41]:
ddd = 'พลอยบงกช' 
print(ddd)

พลอยบงกช


 ข้อความสามารถเป็นภาษาไทยได้ไม่มีปัญหาอะไร


# การเปลี่ยนชนิดของตัวแปร variable casting

In [42]:
aaa + int(ccc)

124548

In [44]:
aaa + float(ccc)

124548.0

In [45]:
(str(aaa))

'1092'

การพิมพ์แบบนี้จะทำได้เฉพาะใน Notebook มันช่วยให้เราดูได้ว่าตอนนี้ค่าที่เราดูหน้าตามันเป็นยังไง ค่ามันคืออะไร

# การดำเนินการ Operation (Operators +,-,*,/,%)

In [46]:
a+b

NameError: ignored

Error นี้บอกว่า ตัวแปร b มันไม่มี แล้วก็บวกไม่ได้

In [47]:
aaa + bbb

1103.0

In [48]:
aaa - bbb

1081.0

In [49]:
ab = aaa*bbb
print(ab)

12012.0


In [50]:
ab + int(ccc)

135468.0

### % เครื่องหมาย modulo

In [51]:
5%3

2

In [52]:
7%3

1

modulo คือการหารเอาเศษ เอาไว้ใช้เช็คว่าตัวเลขเป็นเลขคี่หรือเลขคู่ เอา 2 มาหาร ถ้าหารได้ 0  เป็นเลขคู่ ถ้าหารได้ 1 เป็นเลขคี่

### คำสั่ง print แบบพิเศษ (การ fomat string)

In [53]:
print('ตัวแปร') # สิ่งที่อยู่ข้างในวงเล็บคือ ตัวแปร และ string

ตัวแปร


-print สามารถ print ได้ทั้งตัวแปร และstring

-เพื่อความสวยงามสามารถแก้ไขตัว print ให้ฉลาดมากขึ้นโดยการเพิ่ม f สามารถเอา code ใส่เข้าไปในตัวข้อความได้

เครื่องหมาย {} จะแสดงแค่คำตอบของ 7%3 คือ เท่ากับ 1

In [54]:
print(f'%คือการหารเอาเศษ เช่น 7%3 = {7%3}') # เพิ่ม f หน้า 'string' และใช้ {} ใส่ code 

%คือการหารเอาเศษ เช่น 7%3 = 1


In [55]:
print (f'% คือการหารเอาเศษ เช่น 7%3 ={7%3}\
แต่\n/ คือการหารปกติ เช่น 7/3 = {7/3}\
และ\n// คือการหารเอาส่วน เช่น 7/3 = {7/3}\
หรือ\nใช้ int() เพื่อหารเอาส่วน เช่น int(7/3) = {int(7/3)}') # \n คือการขึ้นบรรทัดใหม่ที่ comเห็น \ ใช้ในการตัด code แต่ com จะไม่เห็น

% คือการหารเอาเศษ เช่น 7%3 =1แต่
/ คือการหารปกติ เช่น 7/3 = 2.3333333333333335และ
// คือการหารเอาส่วน เช่น 7/3 = 2.3333333333333335หรือ
ใช้ int() เพื่อหารเอาส่วน เช่น int(7/3) = 2


# DATA STRUCTURE (โครงสร้างข้อมูล)

### List คือการเอาข้อมูลหลายๆตัวมาเรียงต่อกัน
list สามารถสร้างได้ 2 แบบ ดังนี้

ข้อมูลสามารถเป็นได้ทั้ง ตัวเลข,ตัวอักษรและตัวแปร สัญลักษณ์ของ list คือ [ ]

เราสามารถจัดกลุ่มตัวแปรหลายๆตัวมารวมอยู่ในที่เก็บเดียวกันได้

In [56]:
[111, 'c' ,'ไข่ดาว', aaa]

[111, 'c', 'ไข่ดาว', 1092]

# แบบที่ 1 > square brackets
สร้างด้วยสัญลักษณ์ของ list

In [57]:
list_a = []  #list [] = list ว่าง
print(list_a)

[]


In [58]:
list_b = [111,'c','ไข่ดาว',aaa]
print(list_b)

[111, 'c', 'ไข่ดาว', 1092]


# ลำดับที่อยู่ใน list มีความสำคัญ (ลำดับใน list เริ่มจาก 0,1,2,...)

In [59]:
list_b[2]  #หมายความว่าอยากได้สมาชิกตัวที่2 และลำดับไม่มีความเปลี่ยนแปลง

'ไข่ดาว'

# แบบที่2
จะบอกว่าตัวแปลเท่ากับ list

In [60]:
list_c = list()
print(list_c)

[]


#append() เพิ่มสมาชิกใน list


.append คือเอาสมาชิกไปต่อท้ายของ list

In [62]:
list_b. append('u')
print(list_b)

[111, 'c', 'ไข่ดาว', 1092, 'u', 'u']


ชี้จากข้างหน้า คือ อยากได้ตัวไหนก็สามารถชี้เอาได้เลย

ชี้จากข้างหลัง คือ ต้องใส่เครื่อง - เข้าไป เช่น [-1]

0 คือ สมาชิกตัวแรก , -1 คือ สมาชิกตัวสุดท้าย

In [63]:
list_b[-2]  #ดังที่อยู่ในวงเล็บหลังจากตัวแปร list เรียก index ใช้สำหรับชี้ข้อมูลใน list (ชี้จากข้างหน้า,ชี้จากข้างหลัง)

'u'

#String > list of charactrs 

คำสั่งที่ใช้กับ list สามารถใช้กับ string

In [64]:
t = 'ploybongkot is cute'
print(t)

ploybongkot is cute


In [65]:
t[1] #หมายความคิดว่า ต้องการทราบ tตัวที่ 1

'l'

In [66]:
t[-4]

'c'

In [67]:
t[3]

'y'

In [68]:
t[6]

'n'

#จบ 5 มค 2021 

# คาบที่ 3 11 มค. 2021

In [69]:
len(t)

19

In [70]:
list_b

[111, 'c', 'ไข่ดาว', 1092, 'u', 'u']

ตัวที่อยู่ข้างใน[ ] เราเรียกว่า index (ตัวชี้)

List slicing สามารถทำได้โดยใช้ colon

In [71]:
print(t)
print(t[7:9])
print(len(t[7:9]))

ploybongkot is cute
gk
2


In [72]:
t[0:19:2] #เริ่มตัวที่ 0 โดยหยิบทุกๆ 2 ตัว จนครบ 19 ตัว

'pobnkti ue'

มันจะบอกว่าเริ่มที่ตัวไหนจบที่ตัวไหน แล้วถ้าเพิ่ม : อีกตัว คือ เริ่มตัวที่ 0 โดยหยิบทุกๆ 2 ตัว

In [73]:
z = [1,2,3,4,5,6,7,8,9,10]
z[1:10:2]

[2, 4, 6, 8, 10]

ถ้าเว้นว่างหน้า : หมายความว่า เริ่มตั้งแต่ตัวเลข (0) หรือตัวแรก

ถ้าเว้นว่างหลัง : หมายความว่า ไปจนถึงตัวสุดท้าย (len(list))



In [74]:
print(t)
print(t[:6]) # : มีข้อพิเศษถ้าเราไม่ใส่ตัวเลข คือเริ่มตัวแรก list โดยมันแทน 0 ให้เลย 
print(t[:10])
print(t[-4:])
print(t[:])

ploybongkot is cute
ploybo
ploybongko
cute
ploybongkot is cute


In [75]:
print(z[::2])

[1, 3, 5, 7, 9]


### เราสามารถเอา list มาต่อได้ด้วย + 

In [76]:
t + '??'

'ploybongkot is cute??'

In [77]:
t + list_b ## ไม่สามารถเอา list ปกติมาต่อกับ string ได้

TypeError: ignored

In [78]:
print(list_a)
print(list_b)
print(list_b + list_a)

[]
[111, 'c', 'ไข่ดาว', 1092, 'u', 'u']
[111, 'c', 'ไข่ดาว', 1092, 'u', 'u']


### การแบ่ง String ตามสัญลักษณ์ที่กำหนด -> split string

In [79]:
t

'ploybongkot is cute'

In [80]:
t.split(' ') #split คือการแบ่งโดยระบุเพื่อการแบ่งวรรค

['ploybongkot', 'is', 'cute']

In [81]:
time = '12:30:15'

In [82]:
time.split(':')

['12', '30', '15']

In [83]:
t_sp = t.split(' ')
print(t_sp)

['ploybongkot', 'is', 'cute']


In [84]:
#วิธีรวมกลับ
print(':'.join(t_sp))

ploybongkot:is:cute


HW คำนวณเวลาเป็นวินาทีต่อไปนี้โดยใช้คำสั่ง split() ช่วย (print ออกมาให้สวยงาม)

12:30:15

13:41:07

12:53:15

00:59:25

11:11:11

16:06:09

21:12:30

10:06:15

exp => 00:01:10 = 70 วินาที

exp => 01:00:01 = 3601 วินาที

In [85]:
q1 = '12:30:15'
q1.split(':')
list_q1 = [12,30,15]
hr = list_q1[0]*3600
mi = list_q1[1]*60
s = list_q1[2]
aq1 = hr+mi+s
print(aq1) 


45015


q1 แปลงได้ 45015 วินาที

In [86]:
q2 = '13:41:07'
q2.split(':')
list_q2 = [13,41,7]
hr2 = list_q2[0]*3600
mi2 = list_q2[1]*60
s2 = list_q2[2]
aq2 = hr2+mi2+s2
print(aq2) 


49267


q2 แปลงได้ 49267 วินาที

In [87]:
q3 = '12:53:15'
q3.split(':')
list_q3 = [12,53,15]
hr3 = list_q3[0]*3600
mi3 = list_q3[1]*60
s3 = list_q3[2]
aq3 = hr3+mi3+s3
print(aq3) 

46395


q3 แปลงได้ 46395 วินาที

In [88]:
q4 = '00:59:25'
q4.split(':')
list_q4 = [00,59,25]
hr4 = list_q4[0]*3600
mi4 = list_q4[1]*60
s4 = list_q4[2]
aq4 = hr4+mi4+s4
print(aq4) 

3565


q4 แปลงได้ 3565 วินาที

In [89]:
q5 = '11:11:11'
q5.split(':')
list_q5 = [11,11,11]
hr5 = list_q5[0]*3600
mi5 = list_q5[1]*60
s5 = list_q5[2]
aq5 = hr5+mi5+s5
print(aq5) 

40271


q5 แปลงได้ 40271 วินาที

In [90]:
q6 = '19:06:09'
q6.split(':')
list_q6 = [16,6,9]
hr6 = list_q6[0]*3600
mi6 = list_q6[1]*60
s6 = list_q6[2]
aq6 = hr6+mi6+s6
print(aq6) 

57969


q6 แปลงได้ 57969 วินาที

In [91]:
q7 = '21:12:30'
q7.split(':')
list_q7 = [21,12,30]
hr7 = list_q7[0]*3600
mi7 = list_q7[1]*60
s7 = list_q7[2]
aq7 = hr7+mi7+s7
print(aq7) 

76350


q7 แปลงได้ 76350 วินาที

In [92]:
q8 = '10:06:15'
q8.split(':')
list_q8 = [10,6,15]
hr8 = list_q8[0]*3600
mi8 = list_q8[1]*60
s8 = list_q8[2]
aq8 = hr8+mi8+s8
print(aq8) 

36375


q8 แปลงได้ 36375 วินาที

เฉลยการบ้าน

1.ตั้งโจทย์เอาไว้ใส่ตัวแปร

2.ทำการ split ก็คือแยก ชั่วโมง นาที วินาที ออกมา

3.เปลี่ยนค่ามันจาก string ให้เป็น int แล้วก็เอามาบวกกัน จากนั้นก็ print ออกมา

In [ ]:
 q1 = '12:30:15'

In [93]:
q1_sp = q1.split(':')
print(q1_sp)

['12', '30', '15']


In [100]:
answer1 = int(q1_sp[0])*3600 + int(q1_sp[1])*60 + int(q1_sp[2])
print(answer1)

45015


In [101]:
print(f'เวลา {q1} คำนวณเป็นจำนวนวินาทีได้เท่ากับ {answer1}')

เวลา 12:30:15 คำนวณเป็นจำนวนวินาทีได้เท่ากับ 45015


In [102]:
q1 = '12:30:15'
q1_sp = q1.split(':')
answer1 = int(q1_sp[0])*3600 + int(q1_sp[1])*60 + int(q1_sp[2])
print(f'เวลา {q1} คำนวณเป็นจำนวนวินาทีได้เท่ากับ {answer1}')

เวลา 12:30:15 คำนวณเป็นจำนวนวินาทีได้เท่ากับ 45015
